### DISCUSSION TITLE
Safe killing of training (especially on multi-GPU scenarios)

### DISCUSSION THREAD

**Discussion by flipvox** [espnet/espnet#3735] [Q&A]
I haven't found any discussion nor documentation regarding safe killing of processes, especially during training.  Can someone give some ideas on the safest way to do it?

From my experience, I tried doing a batch kill (e.g. kill $(ps aux |... ) of all relevant processes, but I've learned that python's multiprocessing leaves zombie processes that messed up things when I restart training.

**Comment by zzxiang** [comment]
I usually restart the machine to kill the training...

**Comment by maxwellzh** [comment]
Kill all processes spawned by multiprocess with:
```bash
ps aux | grep $USER | grep python | grep multiprocess | awk '{print $2}'| xargs kill -9
```
